results/cmd/watcher/main.go

```diff
@@ -108,12 +113,58 @@ func (r *reconciler) Reconcile(ctx context.Context, key string) error {
 		r.logger.Errorf("Error converting to proto: %v", err)
 		return err
 	}
-	if _, err := r.client.CreateTaskRunResult(ctx, &pb.CreateTaskRunRequest{
-		TaskRun: p,
-	}); err != nil {
-		r.logger.Error("Error updating TaskRun %s: %v", name, err)
-		return err
-	}
 
+	// Create a TaskRun if it does not exist in results server, update existing one otherwise.
+	annotation := p.GetMetadata().GetAnnotations()
+	if val, ok := annotation["results_id"]; ok {
+		if _, err := r.client.UpdateTaskRunResult(ctx, &pb.UpdateTaskRunRequest{
+			TaskRun:   p,
+			ResultsId: val,
+		}); err != nil {
+			r.logger.Error("Error updating TaskRun %s: %v", name, err)
+			return err
+		}
+		r.logger.Infof("update id ", val)
+	} else {
+		res, err := r.client.CreateTaskRunResult(ctx, &pb.CreateTaskRunRequest{
+			TaskRun: p,
+		})
+		if err != nil {
+			r.logger.Error("Error creating TaskRun %s: %v", name, err)
+			return err
+		}
+
+		// Connect to gke cluster and add results_id to annotations field of TaskRun
+		loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
+		configOverrides := &clientcmd.ConfigOverrides{}
+		kubeConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loadingRules, configOverrides)
+		config, err := kubeConfig.ClientConfig()
+		if err != nil {
+			r.logger.Error("Error creating client config for TaskRun %s: %v", name, err)
+		}
+		client, err := taskinterface.NewForConfig(config)
+		if err != nil {
+			r.logger.Error("Error creating client for TaskRun %s: %v", name, err)
+		}
+		path, err := annotationPath(res.GetResultsId())
+		if err != nil {
+			r.logger.Error("Error jsonpatch for TaskRun %s : %v", name, err)
+		}
+		client.TaskRuns(namespace).Patch(name, types.JSONPatchType, path)
+	}
 	return nil
 }
+
+// AnnotationPath creates a jsonpatch path used for adding results_id to TaskRun annotations field.
+func annotationPath(val string) ([]byte, error) {
+	patches := []jsonpatch.JsonPatchOperation{{
+		Operation: "add",
+		Path:      "/metadata/annotations/results_id",
+		Value:     val,
+	}}
+	patchBytes, err := json.Marshal(patches)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal patch bytes in order to cancel: %v", err)
+	}
+	return patchBytes, nil
```

Comment:
```suggestion
	return json.Marshal(patches)
```

Reply:
done

Reply:
Doesn't look like this change was applied.